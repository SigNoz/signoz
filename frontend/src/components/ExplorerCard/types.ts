import { FormInstance } from 'antd';
import { NotificationInstance } from 'antd/es/notification/interface';
import { AxiosResponse } from 'axios';
import { PANEL_TYPES } from 'constants/queryBuilder';
import { UseMutateAsyncFunction } from 'react-query';
import { ICompositeMetricQuery } from 'types/api/alerts/compositeQuery';
import { Query } from 'types/api/queryBuilder/queryBuilderData';
import {
	DeleteViewPayloadProps,
	SaveViewPayloadProps,
	SaveViewProps,
	ViewProps,
} from 'types/api/saveViews/types';
import { DataSource, QueryBuilderContextType } from 'types/common/queryBuilder';

export interface ExplorerCardProps {
	sourcepage: DataSource;
	children: React.ReactNode;
}

export type GetViewDetailsUsingViewKey = (
	viewKey: string,
	data: ViewProps[] | undefined,
) =>
	| {
			query: Query;
			name: string;
			uuid: string;
			panelType: PANEL_TYPES;
			extraData?: string;
	  }
	| undefined;

export interface IsQueryUpdatedInViewProps {
	viewKey: string;
	data: ViewProps[] | undefined;
	stagedQuery: Query | null;
	currentPanelType: PANEL_TYPES | null;
}

export interface SaveViewWithNameProps {
	sourcePage: ExplorerCardProps['sourcepage'];
	handlePopOverClose: VoidFunction;
	refetchAllView: VoidFunction;
}

export interface SaveViewFormProps {
	viewName: string;
}

export interface MenuItemLabelGeneratorProps {
	viewName: string;
	viewKey: string;
	createdBy: string;
	uuid: string;
	viewData: ViewProps[];
	refetchAllView: VoidFunction;
	sourcePage: ExplorerCardProps['sourcepage'];
}

export interface SaveViewHandlerProps {
	viewName: string;
	compositeQuery: ICompositeMetricQuery;
	sourcePage: ExplorerCardProps['sourcepage'];
	extraData: string;
	panelType: PANEL_TYPES | null;
	notifications: NotificationInstance;
	refetchAllView: SaveViewWithNameProps['refetchAllView'];
	saveViewAsync: UseMutateAsyncFunction<
		AxiosResponse<SaveViewPayloadProps>,
		Error,
		SaveViewProps,
		SaveViewPayloadProps
	>;
	handlePopOverClose: SaveViewWithNameProps['handlePopOverClose'];
	redirectWithQueryBuilderData: QueryBuilderContextType['redirectWithQueryBuilderData'];
	form: FormInstance<SaveViewFormProps>;
}

export interface DeleteViewHandlerProps {
	deleteViewAsync: UseMutateAsyncFunction<DeleteViewPayloadProps, Error, string>;
	refetchAllView: MenuItemLabelGeneratorProps['refetchAllView'];
	redirectWithQueryBuilderData: QueryBuilderContextType['redirectWithQueryBuilderData'];
	notifications: NotificationInstance;
	panelType: PANEL_TYPES | null;
	viewKey: string;
	viewId: string;
	updateAllQueriesOperators: QueryBuilderContextType['updateAllQueriesOperators'];
	sourcePage: ExplorerCardProps['sourcepage'];
}
