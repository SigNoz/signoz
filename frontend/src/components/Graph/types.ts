import {
	ActiveElement,
	Chart,
	ChartData,
	ChartEvent,
	ChartOptions,
	ChartType,
	TimeUnit,
} from 'chart.js';
import { ForwardedRef, ReactNode } from 'react';

import {
	dragSelectPluginId,
	DragSelectPluginOptions,
} from './Plugin/DragSelect';
import {
	intersectionCursorPluginId,
	IntersectionCursorPluginOptions,
} from './Plugin/IntersectionCursor';

export interface StaticLineProps {
	yMin: number | undefined;
	yMax: number | undefined;
	borderColor: string;
	borderWidth: number;
	lineText: string;
	textColor: string;
}

export type GraphOnClickHandler = (
	event: ChartEvent,
	elements: ActiveElement[],
	chart: Chart,
	data: ChartData,
) => void;

export type ToggleGraphProps = {
	toggleGraph(graphIndex: number, isVisible: boolean, reference?: string): void;
};

export type CustomChartOptions = ChartOptions & {
	plugins: {
		[dragSelectPluginId]: DragSelectPluginOptions | false;
		[intersectionCursorPluginId]: IntersectionCursorPluginOptions | false;
	};
};

export interface GraphProps {
	animate?: boolean;
	type: ChartType;
	data: Chart['data'];
	title?: ReactNode;
	isStacked?: boolean;
	onClickHandler?: GraphOnClickHandler;
	name: string;
	yAxisUnit?: string;
	forceReRender?: boolean | null | number;
	staticLine?: StaticLineProps | undefined;
	containerHeight?: string | number;
	onDragSelect?: (start: number, end: number) => void;
	dragSelectColor?: string;
	minTime?: number;
	maxTime?: number;
	ref?: ForwardedRef<ToggleGraphProps | undefined>;
}

export interface IAxisTimeUintConfig {
	unitName: TimeUnit;
	multiplier: number;
}

export interface IAxisTimeConfig {
	unitName: TimeUnit;
	stepSize: number;
}

export interface ITimeRange {
	minTime: number | null;
	maxTime: number | null;
}
