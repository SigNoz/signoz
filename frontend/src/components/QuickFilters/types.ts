import { BaseAutocompleteData } from 'types/api/queryBuilder/queryAutocompleteResponse';
import { Query } from 'types/api/queryBuilder/queryBuilderData';
import { DataSource } from 'types/common/queryBuilder';

export enum FiltersType {
	SLIDER = 'SLIDER',
	CHECKBOX = 'CHECKBOX',
}

export enum MinMax {
	MIN = 'MIN',
	MAX = 'MAX',
}

export enum SpecficFilterOperations {
	ALL = 'ALL',
	ONLY = 'ONLY',
}

export interface IQuickFiltersConfig {
	type: FiltersType;
	title: string;
	attributeKey: BaseAutocompleteData;
	aggregateOperator?: string;
	aggregateAttribute?: string;
	dataSource?: DataSource;
	customRendererForValue?: (value: string) => JSX.Element;
	defaultOpen: boolean;
}

export interface IQuickFiltersProps {
	config: IQuickFiltersConfig[];
	handleFilterVisibilityChange: () => void;
	source: QuickFiltersSource;
	onFilterChange?: (query: Query) => void;
}

export enum QuickFiltersSource {
	LOGS_EXPLORER = 'logs-explorer',
	INFRA_MONITORING = 'infra-monitoring',
	TRACES_EXPLORER = 'traces-explorer',
}
