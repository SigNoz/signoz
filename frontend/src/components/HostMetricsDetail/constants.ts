export enum VIEWS {
	METRICS = 'metrics',
	LOGS = 'logs',
	TRACES = 'traces',
	CONTAINERS = 'containers',
	PROCESSES = 'processes',
	EVENTS = 'events',
}

export const VIEW_TYPES = {
	METRICS: VIEWS.METRICS,
	LOGS: VIEWS.LOGS,
	TRACES: VIEWS.TRACES,
	CONTAINERS: VIEWS.CONTAINERS,
	PROCESSES: VIEWS.PROCESSES,
	EVENTS: VIEWS.EVENTS,
};
