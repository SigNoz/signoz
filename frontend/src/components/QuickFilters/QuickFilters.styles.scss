.quick-filters {
	display: flex;
	flex-direction: column;
	height: 100%;
	border-right: 1px solid var(--bg-slate-400);

	.header {
		display: flex;
		align-items: center;
		justify-content: space-between;
		padding: 10.5px;
		border-bottom: 1px solid var(--bg-slate-400);

		.left-actions {
			display: flex;
			align-items: center;
			gap: 6px;

			.text {
				color: var(--bg-vanilla-400);
				font-family: Inter;
				font-size: 14px;
				font-style: normal;
				font-weight: 400;
				line-height: 18px;
				letter-spacing: -0.07px;
			}

			.sync-tag {
				display: flex;
				padding: 5px 9px;
				flex-direction: column;
				justify-content: center;
				align-items: center;
				gap: 10px;
				border-radius: 2px;
				border: 1px solid rgba(78, 116, 248, 0.2);
				background: rgba(78, 116, 248, 0.1);
				color: var(--bg-robin-500);
				font-family: 'Geist Mono';
				font-size: 14px;
				font-style: normal;
				font-weight: 400;
				line-height: 18px; /* 128.571% */
				text-transform: uppercase;
			}
		}

		.right-actions {
			display: flex;
			align-items: center;
			gap: 12px;

			.divider-filter {
				width: 1px;
				height: 14px;
				background: #161922;
			}

			.sync-icon {
				background-color: var(--bg-ink-500);
				border: 0;
				box-shadow: none;
			}
		}
	}
}

.lightMode {
	.quick-filters {
		background-color: var(--bg-vanilla-100);
		border-right: 1px solid var(--bg-vanilla-300);

		.header {
			border-bottom: 1px solid var(--bg-vanilla-300);

			.left-actions {
				.text {
					color: var(--bg-ink-400);
				}

				.sync-icon {
					background-color: var(--bg-vanilla-100);
				}
			}
			.right-actions {
				.sync-icon {
					background-color: var(--bg-vanilla-100);
				}
			}
		}
	}
}
