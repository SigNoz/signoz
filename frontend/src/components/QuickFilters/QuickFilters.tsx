import './QuickFilters.styles.scss';

import {
	FilterOutlined,
	SyncOutlined,
	VerticalAlignTopOutlined,
} from '@ant-design/icons';
import { Tooltip, Typography } from 'antd';
import { useQueryBuilder } from 'hooks/queryBuilder/useQueryBuilder';
import { cloneDeep } from 'lodash-es';
import { BaseAutocompleteData } from 'types/api/queryBuilder/queryAutocompleteResponse';
import { Query } from 'types/api/queryBuilder/queryBuilderData';

import Checkbox from './FilterRenderers/Checkbox/Checkbox';
import Slider from './FilterRenderers/Slider/Slider';

export enum FiltersType {
	SLIDER = 'SLIDER',
	CHECKBOX = 'CHECKBOX',
}

export enum MinMax {
	MIN = 'MIN',
	MAX = 'MAX',
}

export enum SpecficFilterOperations {
	ALL = 'ALL',
	ONLY = 'ONLY',
}

export interface IQuickFiltersConfig {
	type: FiltersType;
	title: string;
	attributeKey: BaseAutocompleteData;
	customRendererForValue?: (value: string) => JSX.Element;
	defaultOpen: boolean;
}

interface IQuickFiltersProps {
	config: IQuickFiltersConfig[];
	handleFilterVisibilityChange: () => void;
}

export default function QuickFilters(props: IQuickFiltersProps): JSX.Element {
	const { config, handleFilterVisibilityChange } = props;

	const {
		currentQuery,
		lastUsedQuery,
		redirectWithQueryBuilderData,
	} = useQueryBuilder();

	// clear all the filters for the query which is in sync with filters
	const handleReset = (): void => {
		const updatedQuery = cloneDeep(
			currentQuery?.builder.queryData?.[lastUsedQuery || 0],
		);

		if (!updatedQuery) {
			return;
		}

		if (updatedQuery?.filters?.items) {
			updatedQuery.filters.items = [];
		}

		const preparedQuery: Query = {
			...currentQuery,
			builder: {
				...currentQuery.builder,
				queryData: currentQuery.builder.queryData.map((item, idx) => ({
					...item,
					filters: {
						...item.filters,
						items: idx === lastUsedQuery ? [] : [...item.filters.items],
					},
				})),
			},
		};
		redirectWithQueryBuilderData(preparedQuery);
	};

	const lastQueryName =
		currentQuery.builder.queryData?.[lastUsedQuery || 0]?.queryName;
	return (
		<div className="quick-filters">
			<section className="header">
				<section className="left-actions">
					<FilterOutlined />
					<Typography.Text className="text">Filters for</Typography.Text>
					<Tooltip title={`Filter currently in sync with query ${lastQueryName}`}>
						<Typography.Text className="sync-tag">{lastQueryName}</Typography.Text>
					</Tooltip>
				</section>
				<section className="right-actions">
					<Tooltip title="Reset All">
						<SyncOutlined className="sync-icon" onClick={handleReset} />
					</Tooltip>
					<div className="divider-filter" />
					<Tooltip title="Collapse Filters">
						<VerticalAlignTopOutlined
							rotate={270}
							onClick={handleFilterVisibilityChange}
						/>
					</Tooltip>
				</section>
			</section>

			<section className="filters">
				{config.map((filter) => {
					switch (filter.type) {
						case FiltersType.CHECKBOX:
							return <Checkbox filter={filter} />;
						case FiltersType.SLIDER:
							return <Slider filter={filter} />;
						default:
							return <Checkbox filter={filter} />;
					}
				})}
			</section>
		</div>
	);
}
