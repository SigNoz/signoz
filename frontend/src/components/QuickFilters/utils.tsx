import { Filter as FilterType } from 'types/api/quickFilters/getCustomFilters';

import { FiltersType, IQuickFiltersConfig } from './types';

const getFilterName = (str: string): string =>
	// replace . and _ with space
	// capitalize the first letter of each word
	str
		.replace(/\./g, ' ')
		.replace(/_/g, ' ')
		.split(' ')
		.map((word) => word.charAt(0).toUpperCase() + word.slice(1))
		.join(' ');

export const getFilterConfig = (
	customFilters?: FilterType[],
	config?: IQuickFiltersConfig[],
): IQuickFiltersConfig[] => {
	if (!customFilters?.length) {
		return config || [];
	}

	return customFilters.map(
		(att, index) =>
			({
				type: FiltersType.CHECKBOX,
				title: getFilterName(att.key),
				attributeKey: {
					id: att.key,
					key: att.key,
					dataType: att.dataType,
					type: att.type,
					isColumn: att.isColumn,
					isJSON: att.isJSON,
				},
				defaultOpen: index === 0,
			} as IQuickFiltersConfig),
	);
};
