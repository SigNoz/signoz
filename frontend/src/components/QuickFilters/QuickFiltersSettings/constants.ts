import { SignalType } from 'components/QuickFilters/types';
import { DataSource } from 'types/common/queryBuilder';

export const SIGNAL_DATA_SOURCE_MAP = {
	[SignalType.LOGS]: DataSource.LOGS,
	[SignalType.TRACES]: DataSource.TRACES,
	[SignalType.EXCEPTIONS]: DataSource.TRACES,
	[SignalType.API_MONITORING]: DataSource.TRACES,
};
