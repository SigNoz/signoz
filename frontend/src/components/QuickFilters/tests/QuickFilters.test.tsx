import '@testing-library/jest-dom';

import { fireEvent, render, screen } from '@testing-library/react';
import { useGetAggregateValues } from 'hooks/queryBuilder/useGetAggregateValues';
import { useQueryBuilder } from 'hooks/queryBuilder/useQueryBuilder';
import MockQueryClientProvider from 'providers/test/MockQueryClientProvider';

import QuickFilters from '../QuickFilters';
import { QuickFiltersSource } from '../types';
import { QuickFiltersConfig } from './constants';

// Mock the useQueryBuilder hook
jest.mock('hooks/queryBuilder/useQueryBuilder', () => ({
	useQueryBuilder: jest.fn(),
}));
// Mock the useGetAggregateValues hook
jest.mock('hooks/queryBuilder/useGetAggregateValues', () => ({
	useGetAggregateValues: jest.fn(),
}));

const handleFilterVisibilityChange = jest.fn();
const redirectWithQueryBuilderData = jest.fn();

function TestQuickFilters(): JSX.Element {
	return (
		<MockQueryClientProvider>
			<QuickFilters
				source={QuickFiltersSource.EXCEPTIONS}
				config={QuickFiltersConfig}
				handleFilterVisibilityChange={handleFilterVisibilityChange}
			/>
		</MockQueryClientProvider>
	);
}

describe('Quick Filters', () => {
	beforeEach(() => {
		// Provide a mock implementation for useQueryBuilder
		(useQueryBuilder as jest.Mock).mockReturnValue({
			currentQuery: {
				builder: {
					queryData: [
						{
							queryName: 'Test Query',
							filters: { items: [{ key: 'test', value: 'value' }] },
						},
					],
				},
			},
			lastUsedQuery: 0,
			redirectWithQueryBuilderData,
		});

		// Provide a mock implementation for useGetAggregateValues
		(useGetAggregateValues as jest.Mock).mockReturnValue({
			data: {
				statusCode: 200,
				error: null,
				message: 'success',
				payload: {
					stringAttributeValues: [
						'mq-kafka',
						'otel-demo',
						'otlp-python',
						'sample-flask',
					],
					numberAttributeValues: null,
					boolAttributeValues: null,
				},
			}, // Mocked API response
			isLoading: false,
		});
	});

	it('renders correctly with default props', () => {
		const { container } = render(<TestQuickFilters />);
		expect(container).toMatchSnapshot();
	});

	it('displays the correct query name in the header', () => {
		render(<TestQuickFilters />);
		expect(screen.getByText('Filters for')).toBeInTheDocument();
		expect(screen.getByText('Test Query')).toBeInTheDocument();
	});

	it('should add filter data to query when checkbox is clicked', () => {
		render(<TestQuickFilters />);
		const checkbox = screen.getByText('mq-kafka');
		fireEvent.click(checkbox);
		expect(redirectWithQueryBuilderData).toHaveBeenCalledWith(
			expect.objectContaining({
				builder: {
					queryData: expect.arrayContaining([
						expect.objectContaining({
							filters: expect.objectContaining({
								items: expect.arrayContaining([
									expect.objectContaining({
										key: expect.objectContaining({
											key: 'deployment.environment',
										}),
										value: 'mq-kafka',
									}),
								]),
							}),
						}),
					]),
				},
			}),
		); // sets composite query param
	});
});
