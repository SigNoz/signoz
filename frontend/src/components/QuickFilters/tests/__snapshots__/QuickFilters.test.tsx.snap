// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quick Filters renders correctly with default props 1`] = `
<div>
  <div
    class="quick-filters"
  >
    <section
      class="header"
    >
      <section
        class="left-actions"
      >
        <span
          aria-label="filter"
          class="anticon anticon-filter"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="filter"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880.1 154H143.9c-24.5 0-39.8 26.7-27.5 48L349 597.4V838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V597.4L907.7 202c12.2-21.3-3.1-48-27.6-48zM603.4 798H420.6V642h182.9v156zm9.6-236.6l-9.5 16.6h-183l-9.5-16.6L212.7 226h598.6L613 561.4z"
            />
          </svg>
        </span>
        <span
          class="ant-typography text css-dev-only-do-not-override-2i2tap"
        >
          Filters for
        </span>
        <span
          class="ant-typography sync-tag css-dev-only-do-not-override-2i2tap"
        >
          Test Query
        </span>
      </section>
      <section
        class="right-actions"
      >
        <span
          aria-label="sync"
          class="anticon anticon-sync sync-icon"
          role="img"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            data-icon="sync"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M168 504.2c1-43.7 10-86.1 26.9-126 17.3-41 42.1-77.7 73.7-109.4S337 212.3 378 195c42.4-17.9 87.4-27 133.9-27s91.5 9.1 133.8 27A341.5 341.5 0 01755 268.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.7 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c0-6.7-7.7-10.5-12.9-6.3l-56.4 44.1C765.8 155.1 646.2 92 511.8 92 282.7 92 96.3 275.6 92 503.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8zm756 7.8h-60c-4.4 0-7.9 3.5-8 7.8-1 43.7-10 86.1-26.9 126-17.3 41-42.1 77.8-73.7 109.4A342.45 342.45 0 01512.1 856a342.24 342.24 0 01-243.2-100.8c-9.9-9.9-19.2-20.4-27.8-31.4l60.2-47a8 8 0 00-3-14.1l-175.7-43c-5-1.2-9.9 2.6-9.9 7.7l-.7 181c0 6.7 7.7 10.5 12.9 6.3l56.4-44.1C258.2 868.9 377.8 932 512.2 932c229.2 0 415.5-183.7 419.8-411.8a8 8 0 00-8-8.2z"
            />
          </svg>
        </span>
        <div
          class="divider-filter"
        />
        <span
          aria-label="vertical-align-top"
          class="anticon anticon-vertical-align-top"
          role="img"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            data-icon="vertical-align-top"
            fill="currentColor"
            focusable="false"
            height="1em"
            style="transform: rotate(270deg);"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M859.9 168H164.1c-4.5 0-8.1 3.6-8.1 8v60c0 4.4 3.6 8 8.1 8h695.8c4.5 0 8.1-3.6 8.1-8v-60c0-4.4-3.6-8-8.1-8zM518.3 355a8 8 0 00-12.6 0l-112 141.7a7.98 7.98 0 006.3 12.9h73.9V848c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V509.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 355z"
            />
          </svg>
        </span>
      </section>
    </section>
    <div
      class="overlay-scrollbar"
      data-overlayscrollbars-initialize="true"
    >
      <div
        data-overlayscrollbars-contents=""
      >
        <section
          class="filters"
        >
          <div
            class="checkbox-filter"
          >
            <section
              class="filter-header-checkbox"
            >
              <section
                class="left-action"
              >
                <svg
                  class="lucide lucide-chevron-down"
                  cursor="pointer"
                  fill="none"
                  height="13"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="13"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
                <span
                  class="ant-typography title css-dev-only-do-not-override-2i2tap"
                >
                  Environment
                </span>
              </section>
              <section
                class="right-action"
              >
                <span
                  class="ant-typography clear-all css-dev-only-do-not-override-2i2tap"
                >
                  Clear All
                </span>
              </section>
            </section>
            <section
              class="search"
            >
              <input
                class="ant-input css-dev-only-do-not-override-2i2tap"
                placeholder="Filter values"
                type="text"
                value=""
              />
            </section>
            <section
              class="values"
            >
              <div
                class="value"
              >
                <label
                  class="ant-checkbox-wrapper ant-checkbox-wrapper-checked check-box css-dev-only-do-not-override-2i2tap"
                >
                  <span
                    class="ant-checkbox ant-wave-target css-dev-only-do-not-override-2i2tap ant-checkbox-checked"
                  >
                    <input
                      checked=""
                      class="ant-checkbox-input"
                      type="checkbox"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                </label>
                <div
                  class="checkbox-value-section"
                >
                  <span
                    class="ant-typography ant-typography-ellipsis ant-typography-single-line ant-typography-ellipsis-single-line value-string css-dev-only-do-not-override-2i2tap"
                  >
                    mq-kafka
                  </span>
                  <button
                    class="ant-btn css-dev-only-do-not-override-2i2tap ant-btn-text only-btn"
                    type="button"
                  >
                    <span>
                      Only
                    </span>
                  </button>
                  <button
                    class="ant-btn css-dev-only-do-not-override-2i2tap ant-btn-text toggle-btn"
                    type="button"
                  >
                    <span>
                      Toggle
                    </span>
                  </button>
                </div>
              </div>
              <div
                class="value"
              >
                <label
                  class="ant-checkbox-wrapper ant-checkbox-wrapper-checked check-box css-dev-only-do-not-override-2i2tap"
                >
                  <span
                    class="ant-checkbox ant-wave-target css-dev-only-do-not-override-2i2tap ant-checkbox-checked"
                  >
                    <input
                      checked=""
                      class="ant-checkbox-input"
                      type="checkbox"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                </label>
                <div
                  class="checkbox-value-section"
                >
                  <span
                    class="ant-typography ant-typography-ellipsis ant-typography-single-line ant-typography-ellipsis-single-line value-string css-dev-only-do-not-override-2i2tap"
                  >
                    otel-demo
                  </span>
                  <button
                    class="ant-btn css-dev-only-do-not-override-2i2tap ant-btn-text only-btn"
                    type="button"
                  >
                    <span>
                      Only
                    </span>
                  </button>
                  <button
                    class="ant-btn css-dev-only-do-not-override-2i2tap ant-btn-text toggle-btn"
                    type="button"
                  >
                    <span>
                      Toggle
                    </span>
                  </button>
                </div>
              </div>
              <div
                class="value"
              >
                <label
                  class="ant-checkbox-wrapper ant-checkbox-wrapper-checked check-box css-dev-only-do-not-override-2i2tap"
                >
                  <span
                    class="ant-checkbox ant-wave-target css-dev-only-do-not-override-2i2tap ant-checkbox-checked"
                  >
                    <input
                      checked=""
                      class="ant-checkbox-input"
                      type="checkbox"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                </label>
                <div
                  class="checkbox-value-section"
                >
                  <span
                    class="ant-typography ant-typography-ellipsis ant-typography-single-line ant-typography-ellipsis-single-line value-string css-dev-only-do-not-override-2i2tap"
                  >
                    otlp-python
                  </span>
                  <button
                    class="ant-btn css-dev-only-do-not-override-2i2tap ant-btn-text only-btn"
                    type="button"
                  >
                    <span>
                      Only
                    </span>
                  </button>
                  <button
                    class="ant-btn css-dev-only-do-not-override-2i2tap ant-btn-text toggle-btn"
                    type="button"
                  >
                    <span>
                      Toggle
                    </span>
                  </button>
                </div>
              </div>
              <div
                class="value"
              >
                <label
                  class="ant-checkbox-wrapper ant-checkbox-wrapper-checked check-box css-dev-only-do-not-override-2i2tap"
                >
                  <span
                    class="ant-checkbox ant-wave-target css-dev-only-do-not-override-2i2tap ant-checkbox-checked"
                  >
                    <input
                      checked=""
                      class="ant-checkbox-input"
                      type="checkbox"
                    />
                    <span
                      class="ant-checkbox-inner"
                    />
                  </span>
                </label>
                <div
                  class="checkbox-value-section"
                >
                  <span
                    class="ant-typography ant-typography-ellipsis ant-typography-single-line ant-typography-ellipsis-single-line value-string css-dev-only-do-not-override-2i2tap"
                  >
                    sample-flask
                  </span>
                  <button
                    class="ant-btn css-dev-only-do-not-override-2i2tap ant-btn-text only-btn"
                    type="button"
                  >
                    <span>
                      Only
                    </span>
                  </button>
                  <button
                    class="ant-btn css-dev-only-do-not-override-2i2tap ant-btn-text toggle-btn"
                    type="button"
                  >
                    <span>
                      Toggle
                    </span>
                  </button>
                </div>
              </div>
            </section>
          </div>
          <div
            class="checkbox-filter"
          >
            <section
              class="filter-header-checkbox"
            >
              <section
                class="left-action"
              >
                <svg
                  class="lucide lucide-chevron-right"
                  cursor="pointer"
                  fill="none"
                  height="13"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="13"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m9 18 6-6-6-6"
                  />
                </svg>
                <span
                  class="ant-typography title css-dev-only-do-not-override-2i2tap"
                >
                  Service Name
                </span>
              </section>
              <section
                class="right-action"
              />
            </section>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
